runtime-sdk-macros/src/evm_derive.rs

```diff
@@ -0,0 +1,467 @@
+        Ok(a) => a,
+        Err(e) => {
+            e.span().unwrap().error(format!("{}", e)).emit();
+            return None;
+        }
+    };
+
+    let mut signature = parse_signature(f.sig.ident.clone(), attr.signature.as_str());
+    signature.convert = attr.convert.is_present();
+    for inp in f.sig.inputs.iter() {
+        match inp {
+            FnArg::Receiver(recv) => {
+                recv.self_token
+                    .span
+                    .unwrap()
+                    .error("Dispatchable methods can't have a receiver")
+                    .emit();
+                return None;
+            }
+            FnArg::Typed(typed) => match *typed.pat {
+                Pat::Ident(_) => continue,
+                _ => {
+                    typed
+                        .colon_token
+                        .span
+                        .unwrap()
+                        .error("Unsupported function input type")
+                        .emit();
+                    return None;
+                }
+            },
+        }
+    }
+    if signature.convert && signature.args.len() != f.sig.inputs.len() - 1 {
+        signature
+            .name
+            .span()
+            .unwrap()
+            .error("Number of function arguments doesn't match EVM signature")
+            .emit();
+        return None;
+    }
+
+    let byte_tokens = signature.selector.iter().map(|&b| quote!(#b));
+    let byte_list = quote!(#(#byte_tokens ,)*);
+    Some((signature, attr, byte_list))
+}
+
+fn generate_method_call(attr: &EvmMethod, sig: &Signature) -> Option<TokenStream> {
+    let mut decoder_list: Vec<TokenStream> = Vec::new();
+    let mut call_args: Vec<Ident> = Vec::new();
+    let mut arg_decls: Vec<TokenStream> = Vec::new();
+    for (i, evm_ty) in sig.args.iter().enumerate() {
+        let arg_name = format_ident!("arg_{}", i);
+        call_args.push(arg_name.clone());
+        match evm_ty.as_str() {
+            "address" => {
+                decoder_list.push(quote!(::ethabi::ParamType::Address));
+                arg_decls.push(quote! {
+                    let #arg_name = decoded_args[#i].clone().into_address().unwrap();
+                });
+            }
+            "uint256" => {
+                decoder_list.push(quote!(::ethabi::ParamType::Uint(256)));
+                let temp_name = format_ident!("{}_uint", arg_name);
+                arg_decls.push(quote! {
+                    let #temp_name = decoded_args[#i].clone().into_uint().unwrap();
+                    if #temp_name.bits() > (u128::BITS as usize) {
+                        return Some(Err(::evm::executor::stack::PrecompileFailure::Error {
+                            exit_status: ::evm::ExitError::Other("integer overflow".into()),
+                        }));
+                    }
+                    let #arg_name = #temp_name.as_u128();
+                });
+            }
+            ty => {
```

Comment:
I guess this is currently enough for ERC20 and we can extend it later if needed.

Reply:
Yup, I implemented only the ones I needed. It should be trivial to add more types as necessary.